circle.yml

```diff
@@ -1,6 +1,83 @@
+# These environment variables must be set in CircleCI UI
+#
+# DOCKERHUB_REPO - docker hub repo, format: <username>/<repo>
+# DOCKER_EMAIL   - login info for docker hub
+# DOCKER_USER
+# DOCKER_PASS
+
 machine:
+  pre:
+    - curl -sSL https://s3.amazonaws.com/circle-downloads/install-circleci-docker.sh | bash -s -- 1.10.0
   node:
     version: 4.3.1
+  services:
+    - docker
+
 dependencies:
+  cache_directories:
+    - "~/docker"
   pre:
     - npm install -g npm@3
+    - sudo apt-get update; sudo apt-get install pigz
+    - sudo pip install tox mozdownload mozinstall
+    - mozdownload --version latest --destination firefox.tar.bz2
+    - mozinstall firefox.tar.bz2
+    - wget -O geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v0.10.0/geckodriver-v0.10.0-linux64.tar.gz
```

Comment:
All these install steps will slow down the build and it looks like they are only needed for the UI Tests. It would be great if we could defer this to a "downstream" build step somehow (if CircleCI supports that). Ideally, we need to see failures as quick as possible from the output of `npm test`. If that succeeds, we could spend more time configuring and running the UI Tests.